src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Machine.class) {
+            return List.of();
+        }
+
+        try {
+            try {
+                return List.of(type.getDeclaredConstructor().newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
+        } catch (InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        if (!machines.isEmpty()) {
+            for (int i = 0; i < machines.size(); i++) {
+                machines.set(i, value);
+            }
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
```

Comment:
The `startWorking` method should also use the generic type `T` for the `machines` parameter to ensure it can accept lists of any subclass of `Machine`. Change the parameter type to `List<? extends T>` if the class is parameterized with `T`.